crates/services/importer/src/importer.rs

```diff
@@ -206,14 +207,13 @@ where
         ),
         err
     )]
-    fn _commit_result<ExecutorDatabase>(
-        &self,
-        result: UncommittedResult<StorageTransaction<ExecutorDatabase>>,
-    ) -> Result<(), Error>
-    where
-        ExecutorDatabase: ports::ExecutorDatabase,
-    {
-        let (result, mut db_tx) = result.into();
+    fn _commit_result(&self, result: UncommittedResult<Changes>) -> Result<(), Error> {
+        let mut guard = self
+            .database
+            .try_lock()
+            .expect("Semaphore prevents from multiple access to teh database");
```

Comment:
I'm using semaphore because it is the name of the type=D. It helps to connect it with the real object that guarantees safety.
<img width="857" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/998c1782-f3f6-47b5-a79e-dffc9ce9bd85">

Reply:
Yes, but (if I am reading this right) this line accesses `Importer::database` which is a `Mutex<D>` on the database, not on the `guard` (`Semaphore`) on the Importer itself.

Reply:
And the statement references to the `guard`. In all places where we call `_commit_result` we call take a lock for the semaphore first=)